### PR TITLE
Migrate Australia holidays to holiday groups and subdivision holidays methods

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Migrate Australia holidays to holiday groups and subdivision holidays methods.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good (`make pre-commit`)
- [x] I've added tests to verify that the new code works and all tests pass locally (`make test`, `make tox`)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/australia.py [hunk 1]
```diff
@@ -38,6 +37,18 @@ class Australia(HolidayBase):
...
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _add_observed(self, dt: date, name: str = None) -> None:
+        if self.observed:
+            name = name if name else self[dt]
+            self._add_holiday(
+                "%s (Observed)" % name, _get_nth_weekday_from(1, MON, dt)
+            )
```
Comment (arkid15r) [inline holidays/countries/australia.py:50]:
```suggestion
        if self.observed:
            self._add_holiday(
                "%s (Observed)" % (name or self[dt]),
                _get_nth_weekday_from(1, MON, dt),
            )
```

File: holidays/countries/australia.py [hunk 2]
```diff
@@ -53,227 +64,317 @@ def _populate(self, year):
         # TODO do more research on history of Aus holidays
 
         # New Year's Day
-        name = "New Year's Day"
-        jan1 = date(year, JAN, 1)
-        self[jan1] = name
-        if self.observed and self._is_weekend(jan1):
-            self[_get_nth_weekday_from(1, MON, jan1)] = name + " (Observed)"
+        jan1 = self._add_new_years_day("New Year's Day")
```
Comment (arkid15r) [inline holidays/countries/australia.py:67]: I believe we use `jan_1` convention elsewhere.

File: holidays/countries/australia.py [hunk 3]
```diff
@@ -53,227 +64,317 @@ def _populate(self, year):
...
+                if self.subdiv == "NSW" and year <= 1945
+                else "Australia Day"
+            )
+            jan26 = self._add_holiday(name, JAN, 26)
+            if year >= 1946 and self._is_weekend(jan26):
+                self._add_observed(jan26)
         elif year >= 1888 and self.subdiv != "SA":
-            name = "Anniversary Day"
-            self[jan26] = name
+            self._add_holiday("Anniversary Day", JAN, 26)
```
Comment (arkid15r) [inline holidays/countries/australia.py:82]: You can override `_add_subdiv_holidays()` and extract this [non-]`SA` case there.
I guess I've only used it for Portugal so far.

**Review by arkid15r — COMMENTED** [review]
Please look into these comments:

File: holidays/countries/australia.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/australia.py]: Done.

**Pull request merged**